### PR TITLE
Clean shutdown of cloud controllers in integration tests

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/108483

```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig cloud-provider

### CODE REVIEW

File: cmd/kube-scheduler/app/testing/testserver.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func StartTestServer(t Logger, customFlags []string) (result TestServer, err err
 		if errCh != nil {
 			err, ok := <-errCh
 			if ok && err != nil {
-				klog.ErrorS(err, "Failed to shutdown test server clearly")
+				klog.ErrorS(err, "Failed to shutdown test server clearly: %v", err)
```
Comment (aojea) [inline cmd/kube-scheduler/app/testing/testserver.go:76]: > > cmd/kube-scheduler/app/testing/testserver.go:76:5: logging function "ErrorS" should not use format specifier "%v" (logcheck)
> 				klog.ErrorS(err, "Failed to shutdown test server clearly: %v", err)
> 				^
> 
> Please review the above warnings. You can
> 

I think that ErrorS already do the formatting, and you don't have to change this

File: staging/src/k8s.io/cloud-provider/controllers/service/controller.go [hunk 2]
```diff
@@ -287,11 +290,15 @@ func (s *Controller) worker(ctx context.Context) {
...
+	for {
+		select {
+		case <-s.nodeSyncCh:
+			klog.V(4).Info("nodeSync has been triggered")
+			s.nodeSyncInternal(ctx, workers)
+		case <-ctx.Done():
+			return
+		}
 	}
-	klog.V(2).Info("s.nodeSyncCh is closed. Exiting nodeSyncLoop")
```
Comment (aojea) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller.go:294]: super nit, but you could keep it,

File: cmd/kube-scheduler/app/testing/testserver.go (hunk 1 above)
Comment (wojtek-t) [reply cmd/kube-scheduler/app/testing/testserver.go]: Fixed

File: staging/src/k8s.io/cloud-provider/controllers/service/controller.go (hunk 2 above)
Comment (wojtek-t) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller.go:294]: I wanted, but I was getting some klog goroutines hanging forever.
So I got rid of it as it was simpler than debugging klog...

**Comment by wojtek-t** [comment]
@aojea - PTAL

File: staging/src/k8s.io/cloud-provider/controllers/service/controller.go (hunk 2 above)
Comment (aojea) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller.go:294]: Yeah, not worthy

**Comment by aojea** [comment]
/retest

And all these timeouts?

**Comment by wojtek-t** [comment]
The node test failures seems suspicious...

/hold

**Comment by wojtek-t** [comment]
@aojea - For now I left only the unification of testing package for kcm and ccm with other components (like kube-apiserver and scheduler).
I will dig deeper into controller changes later, those ones should be safe.

/hold cancel

**Pull request merged**